packages/react-native-bridge/index.js

```diff
@@ -226,7 +226,7 @@ export function useIsConnected() {
 	return { isConnected };
 }
 
-function subscribeConnectionStatus( callback ) {
+export function subscribeConnectionStatus( callback ) {
```

Comment:
An alternative to exporting this utility would be providing an higher-order component for class components. This practice is [fairly prevalent](https://github.com/WordPress/gutenberg/blob/93349ff2ba3010baf4957f575a9a6a8672ec98c5/packages/data/src/components/with-select/index.js) in the Gutenberg project today. 

There may be value in aligning with project practices and avoiding duplicative logic. WDYT? 

```js
/**
 * WordPress dependencies
 */
import { createHigherOrderComponent } from "@wordpress/compose";

/**
 * Internal dependencies
 */
import { useIsConnected } from "./index";

export default createHigherOrderComponent( ( WrappedComponent ) => {
	return ( props ) => {
		const isConnected = useIsConnected();
		return <WrappedComponent { ...props } isConnected={ isConnected } />;
	};
}, 'withIsConnected' );
```

```js
function MyClassComponent( props ) {
	return props.isConnected ? (
		<Text>Connected</Text>
	) : (
		<Text>Not Connected</Text>
	);
}

export default withIsConnected( MyClassComponent );
```

Reply:
I definitely went down the rabbit hole for using a HOC here (and using Context, too). I am not against the HOC at all, but would welcome your feedback on the thought that kept me from implementing a HOC: do you perceive that we (as in, anyone that works on Gutenberg, and particularly Gutenberg Mobile) have a general mission to update class components to function components? I have the understanding that class components would eventually need to be refactored, and that implementing any further HOCs may discourage us from doing so. I am not certain if this is my own bias as I vastly prefer to work with function components and hooks. I do recognize that an HOC would allow us to benefit from code reuse in this case. 👍

Reply:
I do not perceive a HOC created for the purposes of sharing logic abstracted into a Hook to be a negative thing or a "crutch" discouraging migration. I perceive it to be a utility to help the team retain velocity against the highest priorities, whatever that may be at a given time. 

I do believe the project intends to migrate class components to function components. I perceive it is a judgement call as to the appropriate time do so. Generally, I recommend migrating components first in a separate PR before modifying them. 

All that to say, we should use a HOC here. It allows us to reuse logic that we would otherwise have to repeat and ensure we keep in sync to avoid bugs.

Reply:
Added `withIsConnected` as a HOC, and opened a [PR here](https://github.com/WordPress/gutenberg/pull/56966) so that it could be merged independently from this work. I've updated this PR to `withIsConnected`, however. Will note a few other things in the comments.